### PR TITLE
[FEATURE][processing] Export layout atlas as PDF algorithm

### PR DESCRIPTION
**Pull request by nirvn**
## Description

PDF counterpart to the recent export atlas layout to image algorithm.

### CODE REVIEW

**Comment by nirvn** [comment]
Azure failure unrelated.

File: src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+QString QgsLayoutAtlasToPdfAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Exports an atlas layout as a PDF." );
+}
+
+QString QgsLayoutAtlasToPdfAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm outputs an atlas layout as a PDF file.\n\n"
+                      "If a coverage layer is set, the selected layout's atlas settings exposed in this algorithm "
+                      "will be overwritten. In this case, an empty filter or sort by expressing will turn those "
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp:64]:
```suggestion
                      "will be overwritten. In this case, an empty filter or sort by expression will turn those "
```

**Review by nyalldawson — COMMENTED** [review]
There's a LOT of duplicate code here between this algorithm and the existing atlas to image algorithm -- could that code be moved to a common base class or utility function instead?

**Comment by nirvn** [comment]
@nyalldawson , test failures unrelated, OK to merge?

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**